share/completions/flatpak.fish

```diff
@@ -30,9 +31,21 @@ complete -f -c flatpak -n "not __fish_seen_subcommand_from $commands" -a build-i
 complete -f -c flatpak -n "not __fish_seen_subcommand_from $commands" -a build-sign -d 'Sign an application or runtime'
 complete -f -c flatpak -n "not __fish_seen_subcommand_from $commands" -a build-update-repo -d 'Update the summary file in a repository'
 
-complete -f -c flatpak -n "__fish_seen_subcommand_from run" -a "(flatpak list --app | string match -r '\S+')"
-complete -f -c flatpak -n "__fish_seen_subcommand_from info uninstall" -a "(flatpak list | string match -r '\S+')"
-complete -f -c flatpak -n "__fish_seen_subcommand_from remote-ls remote-modify remote-delete" -a "(flatpak remote-list | string match -r '\S+')"
+if test (string join . "$flatversion[1]" "$flatversion[2]") -ge 1.2
+    for line in (flatpak list --app --columns=application,name)
+        set -l info (string split \t "$line")
+        complete -f -c flatpak -n "__fish_seen_subcommand_from run" -a "$info[1]" -d "$info[2]"
+    end
+    for line in (flatpak list --columns=application,name)
+        set -l info (string split \t "$line")
+        complete -f -c flatpak -n "__fish_seen_subcommand_from info uninstall" -a "$info[1]" -d "$info[2]"
+    end
+    complete -f -c flatpak -n "__fish_seen_subcommand_from remote-ls remote-modify remote-delete" -a "(flatpak remote-list --columns=name)"
+else
+    complete -f -c flatpak -n "__fish_seen_subcommand_from run" -a "(flatpak list --app | string match -r '\S+')"
```

Comment:
If you want to trim leading and/or trailing whitespace `string trim` is a bit more explicit.

Reply:
Is this related to the `string match -r '\S+'` bit?

It was like this in the original code and I left it untouched because I'm not sure of how was the `flatpak list` output before columns (and I assumed there was a good reason for using `string match`).

Reply:
Ah, okay. I missed that it was there before. It's fine to leave it like that, then!

Reply:
After further investigation, it used `string trim` before (98b561929fa01d6f264d4fa0accd14790134e2c0) but it was changed due to #4456.